pkg/middleware/stored_session.go

```diff
@@ -131,7 +119,12 @@ func (s *storedSessionLoader) refreshSessionIfNeeded(rw http.ResponseWriter, req
 // refreshSession attempts to refresh the session with the provider
 // and will save the session if it was updated.
 func (s *storedSessionLoader) refreshSession(rw http.ResponseWriter, req *http.Request, session *sessionsapi.SessionState) error {
-	refreshed, err := s.sessionRefresher(req.Context(), session)
+
+	providerSlice, ok := s.options.GetProviderMapValue(session.ProviderID)
```

Comment:
I would rather see a map of providerID to provider interface within the stored session loader here, we don't need these helpers on the options structs and in general are trying to move away from this pattern